File: MajorElement/maj_elem.py
# The Major Element solution in Python by Steven A. Dunn for CodeEval.com

import sys

for line in open(sys.argv[1], 'r'):
  entries = map(int, line.rstrip('\n').split(','))
  L = len(entries)
  counts = {}

  for entry in entries:
    if entry in counts:
      counts[entry] = counts[entry] + 1
    else:
      counts[entry] = 1

  result = -1
  for key in counts:
    if counts[key] > L/2:
      result = key
      break

  if result != -1:
    print result
  else:
    print "None"

File: TheMajorElement/Ruby/maj_elem.rb
# the major element soln in rb for codeeval by steven a dunn

File.open(ARGV[0], 'r').each_line do |line|
    entries = line.chomp.split(',').map(&:to_i)
    l = entries.length
    counts = {}

    entries.each do |entry|
        if counts.include? entry
            counts[entry] = counts[entry] + 1
        else
            counts[entry] = 1
        end
    end

    result = -1
    counts.each do |key, value|
        if value > l/2
            result = key
            break
        end
    end

    if result != -1
        puts result
    else
        puts "None"
    end
end